packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -4450,10 +4450,32 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin, Res
   ///    state restoration.
   @optionalTypeArgs
   Future<T?> push<T extends Object?>(Route<T> route) {
+    _debugCheckUsePagesAPI(route);
```

Comment:
wrap this in an assert? otherwise, we still wasting time do a function call in release mode